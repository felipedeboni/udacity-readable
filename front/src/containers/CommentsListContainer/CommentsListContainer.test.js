import React from 'react';
import ConnectedCommentsListContainer from './CommentsListContainer';

describe('CommentsListContainer', () => {
  it('renders the loading', () => {
    const [comment] = global.mock.comments;
    const { component } = global.shallowWithStore(
      <ConnectedCommentsListContainer
        category="react"
        postId={comment.parentId}
      />,
      { comments: [] }
    );

    expect(component.dive()).toMatchSnapshot();
  });

  it('maps comments correctly', () => {
    const { commentsObject } = global.mock;
    const [comment] = global.mock.comments;
    const { component } = global.shallowWithStore(
      <ConnectedCommentsListContainer
        category="react"
        postId={comment.parentId}
      />,
      { comments: commentsObject }
    );

    expect(component.dive().instance().props.comments).toEqual(
      global.mock.comments
    );
  });
});
