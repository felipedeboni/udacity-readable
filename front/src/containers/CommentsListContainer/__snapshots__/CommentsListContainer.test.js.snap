// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsListContainer renders the loading 1`] = `
<div>
  <CommentsList
    category="react"
    comments={Array []}
    postId="8xf0y6ziyjabvozdd253nd"
  />
</div>
`;
